torch/distributed/distributed_c10d.py

```diff
@@ -1759,15 +1765,27 @@ def broadcast_object_list(object_list, src=0, group=None):
     else:
         object_sizes_tensor = torch.empty(len(object_list), dtype=torch.long)
 
+    # Current device selection.
+    # To preserve backwards compatibility, ``device`` is default to ``None``
+    # in which case we run current logic of device selection, i.e.
+    # ``current_device`` is CUDA if backend is NCCL otherwise CPU device. In the
+    # case it is not ``None`` we move all intermediate tensors to this given
+    # device. See https://github.com/pytorch/pytorch/issues/60062
```

Comment:
> we move all intermediate tensors to this given device

nit: Just as in this [comment](https://github.com/pytorch/pytorch/pull/61305/files#r667955832), I personally feel that `intermediate tensors` may be misleading. It may be better to say `we move the size and object tensors to be broadcasted to this device`.

Reply:
nit: Also, should we leave the Github issue link here, especially if the issue will be closed post-merge? I feel that ideally the documentation here is sufficiently self-explanatory, which I think it already is.

Reply:
SG, done